### PR TITLE
Add integration testing for the migrate command

### PR DESCRIPTION
**Pull request by josephschorr**
Runs each datastore's backing engine or service in a container, and then creates another container with SpiceDB to perform the migration

### CODE REVIEW

File: internal/testserver/datastore/datastore.go [hunk 1]
```diff
@@ -3,22 +3,38 @@ package datastore
 import (
 	"testing"
 
+	"github.com/stretchr/testify/require"
+
 	"github.com/authzed/spicedb/internal/datastore"
 )
 
 // InitFunc initialize a datastore instance from a uri that has been
 // generated from a TestDatastoreBuilder
```
Comment (ecordell) [inline internal/testserver/datastore/datastore.go:12]:
```suggestion
// InitFunc initializes a datastore instance from a uri that has been
// generated from a RunningEngineForTest
```

File: internal/testserver/datastore/datastore.go [hunk 2]
```diff
@@ -3,22 +3,38 @@ package datastore
...
+// requires env vars set to use from an external source (like a container).
+type RunningEngineForTestWithEnvVars interface {
+	RunningEngineForTest
+
+	// ExternalEnvVars are the environment variables to be set.
+	ExternalEnvVars() []string
+}
+
 // TestDatastoreBuilderFunc allows a function to implement TestDatastoreBuilder
 type TestDatastoreBuilderFunc func(t testing.TB, initFunc InitFunc) datastore.Datastore
```
Comment (ecordell) [inline internal/testserver/datastore/datastore.go:39]: I would rename these too to match

File: internal/datastore/mysql/datastore.go [hunk 3]
```diff
@@ -64,6 +66,10 @@ var (
 	sb = sq.StatementBuilder.PlaceholderFormat(sq.Question)
 )
 
+func init() {
+	datastore.Engines = append(datastore.Engines, Engine)
```
Comment (ecordell) [inline internal/datastore/mysql/datastore.go:70]: 👍

File: internal/testserver/datastore/datastore.go (hunk 2 above)
Comment (josephschorr) [reply internal/testserver/datastore/datastore.go]: Removed

**Review by ecordell — APPROVED** [review]
LGTM

**Pull request merged**